### ISSUE TITLE
Loading tests is... so... slow...

### ISSUE THREAD

**Issue by DustinCampbell** [Pilchie/xunit.runner.wpf#24]

**Comment by jaredpar** [comment]
Yeah I think this is because we send the full serialization string over the wire. It's not needed.  We could just send the name.

**Comment by DustinCampbell** [comment]
It could also be related to the collection refiltering and the UI painting.

**Comment by DustinCampbell** [comment]
@CyrusNajmabadi and I looked at this. It's a confluence of several things:

1. Yes, we're serializing more than we need to. We can do less.
2. xUnit's ```TestMessageVisitor``` isn't all that efficient at dispatching messages due to tons of 'as' casts at the JIT level (see https://github.com/xunit/xunit/blob/fe8d78df7cfec55ddab93c29ea96b0032ca998a8/src/common/TestMessageVisitor.cs#L53). We can do better by implementing ```IMessageSink``` ourselves.
3. We're using AppDomains to do test discovery, which is overkill when we've got the worker process. We should just recycle the process.
4. We can do a better job of adding test cases to the UI in batches.
